include/libp2p/protocol/kademlia/impl/peer_routing_table_impl.hpp

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIBP2P_PROTOCOL_KADEMLIA_PEERROUTINGTABLEIMPL
+#define LIBP2P_PROTOCOL_KADEMLIA_PEERROUTINGTABLEIMPL
+
+#include <libp2p/protocol/kademlia/impl/peer_routing_table.hpp>
+
+#include <boost/assert.hpp>
+#include <deque>
+
+#include <libp2p/event/bus.hpp>
+#include <libp2p/peer/identity_manager.hpp>
+#include <libp2p/protocol/common/sublogger.hpp>
+#include <libp2p/protocol/kademlia/config.hpp>
+#include <libp2p/protocol/kademlia/node_id.hpp>
+
+namespace libp2p::protocol::kademlia {
+
+  /**
+   * @class Bucket
+   *
+   * Single bucket which holds peers.
+   *
+   * @see
+   * https://sourcegraph.com/github.com/libp2p/go-libp2p-kbucket@HEAD/-/blob/bucket.go
```

Comment:
better to provide links that are bound to certain commit, so that for example we still can access the file even if this go file is removed from the repo

Reply:
This is artifact of previous implementation. Will be removed.